modules/dnn/src/model.cpp

```diff
@@ -637,6 +637,294 @@ void DetectionModel::detect(InputArray frame, CV_OUT std::vector<int>& classIds,
         CV_Error(Error::StsNotImplemented, "Unknown output layer type: \"" + lastLayer->type + "\"");
 }
 
+class YOLODetectionModel_Impl : public DetectionModel_Impl
+{
+
+    ImagePaddingMode paddingMode;
+    float padValue;
+    float Height, Width;
+
+    public:
+    YOLODetectionModel_Impl() {
+        //nothing
+    }
+
+    void processFrame(InputArray frame, OutputArrayOfArrays outs)
+    {
+        CV_TRACE_FUNCTION();
+        if (size.empty())
+            CV_Error(Error::StsBadSize, "Input size not specified");
+
+        Width = frame.cols();
+        Height = frame.rows();
+
+        Image2BlobParams param;
+        param.scalefactor = scale;
+        param.size = size;
+        param.mean = mean;
+        param.swapRB = swapRB;
+        param.borderValue = padValue;
+        param.paddingmode = paddingMode;
+
+        Mat blob = dnn::blobFromImageWithParams(frame, param);
+        net.setInput(blob);
+
+        net.forward(outs, outNames);
+    }
+
+    void setPaddingValue(const float padValue_){
+        padValue = padValue_;
+    }
+
+    void setPaddingMode(const ImagePaddingMode paddingmode_){
+        if (paddingmode_ == ImagePaddingMode::DNN_PMODE_NULL ||
+            paddingmode_ == ImagePaddingMode::DNN_PMODE_CROP_CENTER ||
+            paddingmode_ == ImagePaddingMode::DNN_PMODE_LETTERBOX){
+            paddingMode = paddingmode_;
+            } else {
+                CV_Error(Error::StsNotImplemented, "Unsupported padding mode");
+            }
+    }
+
+    void setInputParams(double scale_, const Size& size_, const Scalar& mean_,
+                        bool swapRB_, ImagePaddingMode paddingMode_, float padValue_)
+    {
+        size = size_;
+        mean = mean_;
+        scale = Scalar::all(scale_);
+        swapRB = swapRB_;
+        paddingMode = paddingMode_;
+        padValue = padValue_;
+    }
+
+    void boxRescale(CV_OUT std::vector<Rect>& boxes){
```

Comment:
Can you use [blobRectToImageRect](https://github.com/opencv/opencv/pull/24539) by @LaurentBerger?

Reply:
Can be done in a separate PR